### PR TITLE
contacts app implemented using view-model architecture

### PR DESCRIPTION
**Pull request by DavidMeerkatRefaeli**

### CODE REVIEW

File: app/src/main/java/com/example/drefaeli/myusercontacts/MyApplication.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.example.drefaeli.myusercontacts;
```
[the remark this answers is not in the document]
Comment (DavidMeerkatRefaeli) [reply app/src/main/java/com/example/drefaeli/myusercontacts/MyApplication.java]: this is what I did in the beginning, but Alex (who sits next to me) advised against it.
his arguments were:
1. you cannot really mock an application, so there's no value added from injection
2. you have one instance of the app running, and it should be accessible from everywhere

File: app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public class MyContactsProvider {
+    /**
+     * Loads all contacts from phone and returns it in a list
+     * @return
+     */
+    public ArrayList<String> loadContacts() {
+        ArrayList<String> arrayyListContacts = new ArrayList<>();
+        Cursor phones = MyApplication.getApp().getContentResolver().query(ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
+                null, null, null, null);
```
[the remark this answers is not in the document]
Comment (DavidMeerkatRefaeli) [reply app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java]: yes, you're right

File: app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.ArrayList;
+
+public class MyContactsProvider {
+    /**
+     * Loads all contacts from phone and returns it in a list
+     * @return
+     */
+    public ArrayList<String> loadContacts() {
+        ArrayList<String> arrayyListContacts = new ArrayList<>();
+        Cursor phones = MyApplication.getApp().getContentResolver().query(ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
```
[the remark this answers is not in the document]
Comment (DavidMeerkatRefaeli) [reply app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java]: see what I wrote below regarding the MyApplication singleton class

File: app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                null, null, null, null);
+        while (phones.moveToNext()) {
+            String name = phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME));
+            String phoneNumber = phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
+            arrayyListContacts.add(name + " " + phoneNumber);
```
[the remark this answers is not in the document]
Comment (DavidMeerkatRefaeli) [reply app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java]: ok

File: app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.example.drefaeli.myusercontacts;
+
+import android.database.Cursor;
+import android.provider.ContactsContract;
+import java.util.ArrayList;
+
+public class MyContactsProvider {
```
[the remark this answers is not in the document]
Comment (DavidMeerkatRefaeli) [reply app/src/main/java/com/example/drefaeli/myusercontacts/MyContactsProvider.java]: ok

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -1,12 +1,44 @@
...
+            android:layout_marginLeft="16dp"
+            android:layout_marginTop="8dp"
+            android:layout_marginEnd="16dp"
+            android:layout_marginRight="16dp"
+            android:layout_marginBottom="8dp"
+            android:scrollbars="vertical"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/button" />
```
[the remark this answers is not in the document]
Comment (DavidMeerkatRefaeli) [reply app/src/main/res/layout/activity_main.xml]: ok

File: app/src/main/java/com/example/drefaeli/myusercontacts/MyApplication.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yardenGerecht) [reply app/src/main/java/com/example/drefaeli/myusercontacts/MyApplication.java]: so instead inject the activity

File: app/src/main/res/layout/activity_main.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (yardenGerecht) [reply app/src/main/res/layout/activity_main.xml]: ?

**Comment by yardenGerecht** [comment]
F&S&Merge

**Comment by DavidMeerkatRefaeli** [comment]
changed commit message.
rebases and squashed...
now merging

**Pull request merged**